src/sentry/integrations/source_code_management/webhook.py

```diff
@@ -0,0 +1,26 @@
+from abc import ABC, abstractmethod
+from collections.abc import Mapping
+from typing import Any
+
+from sentry.integrations.utils.metrics import IntegrationWebhookEventType
+from sentry.models.repository import Repository
+
+
+class SCMWebhook(ABC):
```

Comment:
This is something I still need to play around with, but for classes like this with no implementation we may be able to use protocols instead of ABCs: https://typing.readthedocs.io/en/latest/spec/protocol.html#defining-a-protocol